pkg/graveler/retention/garbage_collection_manager.go

```diff
@@ -46,6 +48,41 @@ func (m *GarbageCollectionManager) GetAddressesLocation(sn graveler.StorageNames
 	return qk.Format(), nil
 }
 
+func (m *GarbageCollectionManager) GetUncommittedLocation(runID string, sn graveler.StorageNamespace) (string, error) {
+	key := fmt.Sprintf(uncommittedFilePrefixTemplate, m.committedBlockStoragePrefix, runID)
+	qk, err := block.ResolveNamespace(sn.String(), key, block.IdentifierTypeRelative)
+	if err != nil {
+		return "", err
+	}
+	return qk.Format(), nil
+}
+
+func (m *GarbageCollectionManager) SaveGarbageCollectionUncommitted(ctx context.Context, repository *graveler.RepositoryRecord, filename, runID string) error {
+	location, err := m.GetUncommittedLocation(runID, repository.StorageNamespace)
+	if err != nil {
+		return err
+	}
+
+	fd, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	stat, err := fd.Stat()
+	if err != nil {
+		return err
+	}
+
+	identifier := fmt.Sprintf("%s%s", location, filename)
```

Comment:
No `/` to separate?

Reply:
location is already delimited

Reply:
I'll forever be the advocate for double/missing slash bugs.. Can we use `path.Join`?

Reply:
1. Not sure that location must end with '/' based on the API we use - I think we should check if it ends with '/' and add if needed.
2. Using + (`location+filename`) looks and usually perform better over `fmt.Sprintf("%s%s", location, filename)`